### ISSUE TITLE
Bug Report：Missing rows in plot_cpdb_heatmap()

### ISSUE THREAD

**Issue by Tu-feng** [zktuong/ktplotspy#16]
Hello, I'm using plot_cpdb_heatmap() function to visualize my result from CPDB, and found a cell type row(B cell) is missing in the 9x10 heatmap. 
It turns out that B cell in my pvals result have no significant interaction as cpdb target. And I expect a row with 0 interaction rather than just drop it. 

```python
    metadata = adata.obs.copy()
    all_intr = pvals.copy()
    labels = metadata[celltype_key]
    intr_pairs = all_intr.interacting_pair
    all_int = all_intr.iloc[:, 12 : all_intr.shape[1]].T
    all_int.columns = intr_pairs
    all_count = all_int.melt(ignore_index=False).reset_index()
    if degs_analysis:
        all_count = all_count[all_count.value == 1]
    else:
        all_count = all_count[all_count.value < alpha]
    count1x = all_count[["index", "interacting_pair"]].groupby("index").agg({"interacting_pair": "count"})
    tmp = pd.DataFrame([x.split("|") for x in count1x.index])

```
Here is the code, seems it gets "tmp" by the filtered "all_counts", and cell without significant interaction were dropped in the filtered "all_counts".
Thanks for your time!

**Comment by zktuong** [comment]
Hi there, thanks for reporting this.

Let me have a think about this more (unless you have any suggestions) as i have limited bandwidth at the moment - i can forsee a potential problem in that the 0 row/column will fail the hierarchical clustering step handled internally by seaborn.cluster_map.

so perhaps there should be a separate code chunk to handle this, and then pass it to seaborn.heatmap perhaps

have you tried the R version and/or cellphonedb’s native plot function to see if it also returns like this? Pinging @datasome to see if he has any suggestions as well

**Comment by Tu-feng** [comment]
@zktuong 
I tried R version plot_cpdb_heatmap(), and the same problem occurred.  I'll try the native plot later.
I temporarily modified the python fuc, and get the heatmap I expect. No errors were encountered.  Seaborn version is 0.11.2
I didn't get that, do you mean a row/column full of 0 may throw an error in the seaborn.clusterheatmap()?
```python
    if degs_analysis:
        all_count_lim = all_count[all_count.value == 1]
    else:
        all_count_lim = all_count[all_count.value < alpha]
    count1x = all_count[["index", "interacting_pair"]].groupby("index").agg({"interacting_pair": "count"})
    tmp = pd.DataFrame([x.split("|") for x in count1x.index])

    count1x = all_count_lim[["index", "interacting_pair"]].groupby("index").agg({"interacting_pair": "count"})
    count_final = pd.concat([tmp, count1x.reset_index(drop=True)], axis=1)
    count_final.columns = ["SOURCE", "TARGET", "COUNT"]
```

**Comment by zktuong** [comment]
Oh ok. perhaps can you put in a pull request and i can do some tests?

**Comment by zktuong** [comment]
closed by #17
